development/code/ui/InventoriesUI.ts

```diff
@@ -0,0 +1,91 @@
+import { INVENTORY_CLASS_ID, INVENTORY_WIDTH } from '../logic/Constants';
+import { PlayerColors } from '../logic/Players';
+import { Item } from '../logic/items/Items';
+import { dragElement } from './Events';
+
+
+const inventoryElement = document.getElementsByClassName(INVENTORY_CLASS_ID)[0];
+
+function createPlayerInventoryElement(playerColor: PlayerColors): HTMLElement {
+  const playerInventoryElement = document.createElement('div');
+  playerInventoryElement.id = playerColor;
+  playerInventoryElement.classList.add('player-inventory');
+  inventoryElement?.appendChild(playerInventoryElement);
+  
+  return playerInventoryElement;
+}
+
+export function showItemOnInventory(
+  item: Item,
+  playerColor: PlayerColors,
+): HTMLElement | undefined {
+  const inventoryItemElement = document.createElement('div') as HTMLElement;
+  inventoryItemElement.id = item.name;
+  inventoryItemElement.classList.add('item');
+  inventoryItemElement.classList.add('inventory-item');
+  inventoryItemElement.innerHTML = item.resource;
+  inventoryItemElement.draggable = true;
+  
+  
+
+  dragElement(inventoryItemElement);
+
+  const playerInventoryElement = document.getElementById(playerColor);
+
+  playerInventoryElement?.childNodes.forEach((child) => {
+    if (!child.hasChildNodes()) {
+      child.appendChild(inventoryItemElement);
+      return;
+    }
+  });
+  if (playerInventoryElement) {
+    return playerInventoryElement;
+  }
+  return undefined;
+}
+
+export function initialiseInventoryUI(playerColor: PlayerColors) {
+  const playerInventoryElement = createPlayerInventoryElement(playerColor);
+
+  for (let row = 0; row < INVENTORY_WIDTH; row++) {
+    createInventorySlotElement(playerInventoryElement, playerColor);
+  }
+  
+  if (playerColor === PlayerColors.BLACK) {
+    playerInventoryElement?.classList.add('collapsed');
+  }
+}
+
+function createInventorySlotElement(
+  playerInventoryElement: HTMLElement,
+  playerColor: PlayerColors,
+) {
+  const squareElement = document.createElement('div');
+  squareElement.classList.add('inventory-square');
+  squareElement.setAttribute('player-color', playerColor);
+
+  playerInventoryElement.appendChild(squareElement);
+}
+
+export function switchShownInventory(playerColor: PlayerColors): boolean | undefined {
+  const playerInventoryElement = document.getElementById(playerColor);
+  if (!playerInventoryElement) {
+    return;
+  }
```

Comment:
Instead of returning `undefined`, why not return `false` (or `true`? What do you think is more correct?) if the `playerInventoryElement` doesn't exist?